frontend/src/components/SearchPage.js

```diff
@@ -6,15 +6,58 @@ import { Button, Modal } from 'react-bootstrap';
 import MatchingService from "../services/matchingService.js"
 
 const SearchPage = () => {
+    const [showReturnToInterview, setShowReturnToInterview] = useState(false)
     const [showCancelFindMatch, setShowCancelFindMatch] = useState(false)
+    const [timerKey, setTimerKey] = useState(0)
 
     const location = useLocation()
     const navigate = useNavigate()
 
     const difficulty = location.state.difficulty
     const username = location.state.username
 
+    const handleCloseReturnToInterview = () => setShowCancelFindMatch(false)
+    const handleShowReturnToInterview = () => {
+        setShowReturnToInterview(true)
+    }
+    const handleReturnToInterview = () => {
+        MatchingService.getInterviewByUsername(username)
+            .then(res => {
+                console.log("returning to existing interview")
+                navigate("/interview", { state: { interviewId: res.data.interviewId, username: username } })
+            })
+    }
+    const handleDeleteInterview = () => {
+        MatchingService.getInterviewByUsername(username)
+            .then(res => {
+                const interviewId = res.data.interviewId
+                MatchingService.deleteInterview(interviewId)
+                    .then(res => {
+                        console.log("existing interview deleted. starting match now")
+                        setTimerKey(prevKey => prevKey + 1)
+                        findMatch()
+                    })
+            })
+    }
+
     useEffect(() => {
+        MatchingService
+            .getInterviewByUsername(username)
```

Comment:
I think the function in the backend requires a req.body parameter instead.

Reply:
Edited to fix backend to read in the correct params for username